operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventSourceManager.java

```diff
@@ -134,8 +134,8 @@ public final synchronized void registerEventSource(String name, EventSource even
   @SuppressWarnings("unchecked")
   public void broadcastOnResourceEvent(ResourceAction action, R resource, R oldResource) {
     eventSources.additionalNamedEventSources().forEach(eventSource -> {
-      if (eventSource instanceof ResourceEventAware) {
-        var lifecycleAwareES = ((ResourceEventAware<R>) eventSource);
+      if (eventSource.original() instanceof ResourceEventAware) {
```

Comment:
Are there possibly other spots where a similar change would be needed?

Reply:
Well, good question, probably not, since it works. Just this did not have a integration test.